### ISSUE TITLE
Race condition or caching problem

### ISSUE THREAD

**Issue by trial-n-error** [getgrav/grav-plugin-comments#27]
Hi,

I face a strange problem. I installed the plugin and grav (current developer versions) delivers different page content on reloads concerning the comment plugin. The comment block is missing. The rest of the page is ok.

I cleared all caches. I tried on two different host environments (nginx, apache with different php variants). 

So I thought it might be connected to {% if grav.twig.enable %} in the template file. If I remove it I can see that the comment block is always printed on reloads, but now sometimes  entered comments are instead missing from time to time.

It seems to be some sort of race condition.

Any advice what I might try?

Best regards,

Form v1.3.0-rc.4
Comments 1.1.4
Grav 1.1.0-rc.3

**Comment by trial-n-error** [comment]
Issue in my configuration. Issue closed.

**Comment by flaviocopes** [comment]
What was the issue exactly?

**Comment by trial-n-error** [comment]
:)

Let me rephrase the "reload" I did. I actually clicked often the langswitcher flags for the two languages I have. In addition grav gives me freedom for the routes to be configured for articles/posts. I make heavy use of this to keep urls from my previous website engine.

The langswitcher plugin, when it derives the route for the page of a different language uses a "raw route" which might be not the correct one (also in my setup due to the overrides). The plugin will compute the proper route/link for if the language is the active one. So in my case I was on the english page. Clicked german. Comments disappeared. Clicked again german. Comments appeared.

Finally I found out that the url between the two clicks is slightly different. But on the first click route/url the comments were disabled in the configuration. So the comment plugin is correct here. It does what it should do.

I figured out in langswitcher code that it is the rawRoute which generates the first route, checked github and was pointed to https://github.com/getgrav/grav-plugin-langswitcher/issues/4. Solution for me is to use the redirect_default_route option.

Thanks again,
 Mario
